src/soroban.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Soroban helper class
+ * formatting, parsing, and etc
+ * @class Soroban
+ */
+export class Soroban {
+  /**
+   * All arithmetic inside the contract is performed on integers to
+   * avoid potential precision and consistency issues of floating-point
+   *
+   * This function takes the smart contract value and its decimals (if the token has any) and returns a display value
+   * @param {string} amount - the token amount you want to display
+   * @param {number} decimals - specify how many decimal places a token has
+   * @returns {string} - display value
+   */
+  static formatTokenAmount(amount, decimals) {
+    let formatted = amount;
+
+    if (amount.includes('.')) {
+      throw new Error('No decimal is allowed');
+    }
+
+    if (decimals > 0) {
+      formatted = [
+        formatted.slice(0, -decimals),
+        formatted.slice(-decimals)
+      ].join('.');
+    }
+
+    // remove trailing zero if any
+    return formatted.replace(/\.?0+$/, '');
```

Comment:
I think this breaks with non-decimal zeroes due to the `?`, e.g. you should test

```js
formatTokenAmount("1230", 0)
```

I think this will give `123` :fearful: The fix is to require the decimal in the regex.

Reply:
updated